oauth2/oauth2-resource-server/src/main/java/org/springframework/security/oauth2/server/resource/web/BearerTokenAuthenticationFilter.java

```diff
@@ -171,4 +169,15 @@ public void setAuthenticationFailureHandler(final AuthenticationFailureHandler a
 		this.authenticationFailureHandler = authenticationFailureHandler;
 	}
 
+	/**
+	 * Set the {@link BearerTokenAuthenticationConverter} to use. Defaults to
+	 * {@link BearerTokenAuthenticationConverter}.
+	 * @param authenticationConverter the {@code AuthenticationConverter} to use
+	 * @since 5.5
+	 */
+	public void setAuthenticationConverter(BearerTokenAuthenticationConverter authenticationConverter) {
```

Comment:
Let's please use the `AuthenticationConverter` interface here so that applications can switch out the implementation completely.